### PR TITLE
Update default pytest output

### LINKED ISSUE #1026: Update default output formatting for pytest CI
**Problem description**

The changes of CI introduced with #1020 create some overly verbose output even for short tests. While the changes to the progress bar per se are ok (alternative would be to pass `-v` so output resembles old style), some of the timing/logging information should not be displayed by default. Note: `pytest` also respects settings in configuration files.

**Steps to reproduce**

This is one of the shorter tests within the test suite ...

```
docker@xps13:/src$ scons test-python-reaction
scons: Reading SConscript files ...
[...]
============================================== test session starts ===============================================
platform linux -- Python 3.6.9, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: /src, inifile:
collected 33 items                                                                                               

../../build/python/cantera/test/test_reaction.py .................................                         [100%]

--------------------------------- generated xml file: /src/test/work/pytest.xml ----------------------------------
=========================================== slowest 50 test durations ============================================
0.04s setup    build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_duplicate
0.02s call     build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_duplicate
0.01s setup    build/python/cantera/test/test_reaction.py::TestElementary::test_add_rxn
0.00s setup    build/python/cantera/test/test_reaction.py::TestElementaryNew::test_add_rxn
0.00s setup    build/python/cantera/test/test_reaction.py::TestCustom::test_add_rxn
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_add_rxn
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_from_dict
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_no_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_add_rxn
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_replace_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_replace_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_from_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_from_dict
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_from_dict
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_no_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_replace_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_from_parts
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_from_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_from_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_custom
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_from_func
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_add_rxn
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_no_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_implicit_three_body
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_from_parts
0.00s call     build/python/cantera/test/test_reaction.py::TestElementaryNew::test_from_parts
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementary::test_from_dict
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_wrong_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementary::test_from_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestCustom::test_from_dict
0.00s teardown build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_duplicate
0.00s teardown build/python/cantera/test/test_reaction.py::TestCustom::test_from_rate
0.00s call     build/python/cantera/test/test_reaction.py::TestCustom::test_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_implicit_three_body
0.00s call     build/python/cantera/test/test_reaction.py::TestElementary::test_wrong_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_not_three_body
0.00s setup    build/python/cantera/test/test_reaction.py::TestCustom::test_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementaryNew::test_add_rxn
0.00s teardown build/python/cantera/test/test_reaction.py::TestCustom::test_custom
0.00s setup    build/python/cantera/test/test_reaction.py::TestCustom::test_from_dict
0.00s call     build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_short_serialization
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementaryNew::test_from_parts
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementary::test_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementary::test_from_parts
0.00s teardown build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_user_override
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementaryNew::test_from_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementary::test_add_rxn
0.00s teardown build/python/cantera/test/test_reaction.py::TestImplicitThirdBody::test_short_serialization
0.00s teardown build/python/cantera/test/test_reaction.py::TestElementaryNew::test_no_rate
0.00s teardown build/python/cantera/test/test_reaction.py::TestCustom::test_add_rxn
===================================================== PASSES =====================================================
______________________________________ TestImplicitThirdBody.test_duplicate ______________________________________
_________________________________ TestImplicitThirdBody.test_implicit_three_body _________________________________
_________________________________ TestImplicitThirdBody.test_non_integer_stoich __________________________________
___________________________________ TestImplicitThirdBody.test_not_three_body ____________________________________
_________________________________ TestImplicitThirdBody.test_short_serialization _________________________________
____________________________________ TestImplicitThirdBody.test_user_override ____________________________________
__________________________________________ TestElementary.test_add_rxn ___________________________________________
_________________________________________ TestElementary.test_from_dict __________________________________________
_________________________________________ TestElementary.test_from_parts _________________________________________
_________________________________________ TestElementary.test_from_rate __________________________________________
__________________________________________ TestElementary.test_no_rate ___________________________________________
____________________________________________ TestElementary.test_rate ____________________________________________
________________________________________ TestElementary.test_replace_rate ________________________________________
_________________________________________ TestElementary.test_wrong_rate _________________________________________
____________________________________________ TestCustom.test_add_rxn _____________________________________________
_____________________________________________ TestCustom.test_custom _____________________________________________
___________________________________________ TestCustom.test_from_dict ____________________________________________
___________________________________________ TestCustom.test_from_func ____________________________________________
___________________________________________ TestCustom.test_from_parts ___________________________________________
___________________________________________ TestCustom.test_from_rate ____________________________________________
____________________________________________ TestCustom.test_no_rate _____________________________________________
______________________________________________ TestCustom.test_rate ______________________________________________
___________________________________________ TestCustom.test_rate_func ____________________________________________
__________________________________________ TestCustom.test_replace_rate __________________________________________
___________________________________________ TestCustom.test_wrong_rate ___________________________________________
_________________________________________ TestElementaryNew.test_add_rxn _________________________________________
________________________________________ TestElementaryNew.test_from_dict ________________________________________
_______________________________________ TestElementaryNew.test_from_parts ________________________________________
________________________________________ TestElementaryNew.test_from_rate ________________________________________
_________________________________________ TestElementaryNew.test_no_rate _________________________________________
__________________________________________ TestElementaryNew.test_rate ___________________________________________
______________________________________ TestElementaryNew.test_replace_rate _______________________________________
_______________________________________ TestElementaryNew.test_wrong_rate ________________________________________
=========================================== 33 passed in 0.12 seconds ============================================
scons: done building targets.

```

**System information**

- Cantera version: 2.6.0a2

**Other thoughts**

I should have noticed this earlier ...

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

- Switch `pytest` output to verbose (`-v` option) 
- Hide information on longest runs by default
- Implement boolean SCons variable `show_long_tests`

The rationale for switching to verbose output is that segfaults are difficult to locate with the progress bar. Further, the longest running tests can still be displayed using
```
scons test-python-reaction show_long_tests=yes
```

While `test_convert.py` (`scons test-python-convert`) creates numerous `WARNING` messages from captured log calls - which may confuse some users - I am leaving those for another PR as those may or may not be a feature.

**If applicable, fill in the issue number this pull request is fixing**

Fixes #1026

**Checklist**

- [x] There is a clear use-case for this code change
- [x] The commit message has a short title & references relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: test/SConscript [hunk 1]
```diff
@@ -3,6 +3,7 @@ import subprocess
 from xml.etree import ElementTree
 from os.path import join as pjoin
 import time
+import sys
```
Comment (bryanwweber) [inline test/SConscript:6]: Is this import used in the file?

**Review by bryanwweber — CHANGES_REQUESTED** [review]
Thanks @ischoegl This looks good to me. Just one small question.

File: test/SConscript (hunk 1 above)
Comment (ischoegl) [reply test/SConscript:6]: It is - stumbled across this (rarely encountered) glitch when I ran into an exception. Link is here: https://github.com/Cantera/cantera/blob/33a2e87866be22ee08e6b89567e230f3a5627dc6/test/SConscript#L68-L71
(there are actually three of those)

**Review by bryanwweber — DISMISSED** [review]

**Comment by bryanwweber** [comment]
I'll wait for @speth to review, since he was specifically requested.

**Comment by ischoegl** [comment]
@bryanwweber and @speth ... I updated the interface so the verbose output is optional, e.g.
```
scons test-python-reaction verbose_tests=yes
```
So, the changes are as follows:
* verbose output is *not* shown by default (keeping existing behavior), but can be turned on using `verbose_tests=yes`
* long tests are *not* shown by default (changing existing behavior), but can be turned on using `show_long_tests=yes`

One caveat is that verbosity only affects `pytest`; I found some external work on [googletest verbosity](https://gist.github.com/elliotchance/8215283), but that goes well beyond the scope of what I want to do here as I'm not familiar with the setup of these runners.

**Comment by ischoegl** [comment]
Introducing the `verbose_tests` flag has the advantage that it allows for disabling logger `WARNING` displays by default, which may be confusing to many especially for `test_python_convert.py`. The default display no longer shows warnings, I.e.:
```
docker@xps13:/src$ scons test-python-convert                 
[...]

* INFO: using Cantera module found at this location:
*      '/src/build/python/cantera/__init__.py'
* INFO: Cantera version: 2.6.0a2
* INFO: Git commit: b8a692300 

============================= test session starts =============================
platform linux -- Python 3.6.9, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /src
collected 141 items                                                           

../../build/python/cantera/test/test_convert.py ....................... [ 16%]
....................................................................... [ 66%]
...............................................                         [100%]

=================================== PASSES ====================================
---------------- generated xml file: /src/test/work/pytest.xml ----------------
============================ 141 passed in 46.47s =============================
scons: done building targets.
docker@xps13:/src$ 
```
The `WARNING` messages will be displayed if `verbose_tests=yes`.

**Review by speth — CHANGES_REQUESTED** [review]
I think the implementation of the new options here is fine.

Given that the initial purpose of printing out test durations was as a diagnostic for the CI runs, I think both `show_long_test` and `verbose_tests` should be set to `y` for all CI runs.

**Comment by ischoegl** [comment]
@speth ... thank you for your feedback! Since I initially looked at this, I have started to to believe that there is a benefit to having shorter output by default for users that compile Cantera from source. For development, it is a different issue, where we often need more detailed information, but the flags are easy to toggle.

> Given that the initial purpose of printing out test durations was as a diagnostic for the CI runs, I think both `show_long_tests` and `verbose_tests` should be set to `y` for all CI runs. 

As an alternative to changing the default (where suppressing detailed output imho makes sense), I just flipped the flags for GH actions work flows. What do you think?

**Review by speth — APPROVED** [review]
> As an alternative to changing the default (where suppressing detailed output imho makes sense), I just flipped the flags for GH actions work flows. What do you think?

Yes, that's exactly what I was suggesting. Sorry if that wasn't clear. With this update, I think this is good.

**Pull request merged**